src/examples/follow_trajectory_action_client.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import json
+import rclpy
+from rclpy.action import ActionClient
+from rclpy.node import Node
+
+from std_msgs.msg import Header
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+from builtin_interfaces.msg import Duration
+from control_msgs.action import FollowJointTrajectory
+from trajectory_msgs.msg import JointTrajectory
+
+from ament_index_python.packages import get_package_share_directory
+
+class FollowJointTrajectoryActionClient(Node):
+
+    def __init__(self):
+        super().__init__('follow_joint_trajectory_client', allow_undeclared_parameters = True, automatically_declare_parameters_from_overrides = True)
+        self._action_client = ActionClient(self, FollowJointTrajectory, '/arm_1/arm_controller/follow_joint_trajectory')
+        
+        path = get_package_share_directory('youbot_driver_ros_interface')+'/../../lib/youbot_driver_ros_interface/trajectory.json'
+
+        f = open(path)
+        self.data = json.load(f)
+
+        
+    def send_goal(self):
+
+        header = Header()
+
+        header.stamp = self.get_clock().now().to_msg()
+        header.frame_id = 'arm_link_0'
+    
+        goal_msg = FollowJointTrajectory.Goal()
+        
+        goal_msg.trajectory = JointTrajectory()
+
```

Comment:
since a pre-condition is that the arm is in the 'folded' configuration, can you add a check here to make sure?

Reply:
Sure